### PR TITLE
fix local gw upgrade jobs for ovnk presubmits

### PR DESCRIPTION
**Pull request by jluhrsen**
- 4.15->4.16 job file was misnamed and ended up running on master PRs it's named properly now
- 4.16->4.17 job was missing it's added now
- golang and builder versions were wrong in the 4.15->4.16 job moved to go1.21 and 4.16

### CODE REVIEW

**Comment by trozet** [comment]
/lgtm

thanks @jluhrsen !

File: ci-operator/config/openshift/ovn-kubernetes/openshift-ovn-kubernetes-release-4.17__4.17-upgrade-from-stable-4.16-local-gateway.yaml [hunk 1]
```diff
@@ -62,7 +62,7 @@ tests:
       TEST_TYPE: upgrade-conformance
     workflow: openshift-upgrade-aws-ovn
 zz_generated_metadata:
-  branch: master
+  branch: release-4.17
   org: openshift
   repo: ovn-kubernetes
-  variant: 4.16-upgrade-from-stable-4.15-local-gateway
+  variant: 4.17-upgrade-from-stable-4.16-local-gateway
```
Comment (tssurya) [inline ci-operator/config/openshift/ovn-kubernetes/openshift-ovn-kubernetes-release-4.17__4.17-upgrade-from-stable-4.16-local-gateway.yaml:68]: the name of this file still says 4.16->4.17

**Pull request merged**